packages/language-server/src/lib/Schema.ts

```diff
@@ -1,30 +1,32 @@
 import { TextDocument } from 'vscode-languageserver-textdocument'
 
-export type Line = [document: SchemaDocument, lineNo: number, text: string]
+export type Line = {
+  readonly document: SchemaDocument
+  readonly lineIndex: number
+  readonly text: string
+  readonly untrimmedText: string
```

Comment:
it does feel a little silly to have both of these but I think just having one would be a bigger refactor which would be out of scope for this pr

Reply:
Yes, for the most part we actually need trimmed text, except few places where we need trimmed